react-1/src/components/User.js

```diff
@@ -0,0 +1,119 @@
+import React, {useImperativeHandle, useRef} from 'react';
+import { DragSource, DropTarget } from 'react-dnd';
+import {createUseStyles} from 'react-jss';
+import { useSpring, animated } from 'react-spring';
+import ItemTypes from './ItemTypes';
+
+const useStyles = createUseStyles({
+    User: {
+        minHeight: 40,
+        padding: 40,
+        backgroundColor: '#fff',
+        color: '#949494',
+        cursor: 'move'
+    },
+	Container: {
+		borderBottom: '1px solid #e4e4e4',
+		"&:last-child": {
+			borderBottom: 'inherit'
+		},
+	},
+    Infos: {float: 'left'},
+    UserName: {
+        fontWeight: 'bold',
+        fontSize: 20,
+        float: 'left',
+        marginBottom: 5
+    },
+    UserEmail: {
+        textDecoration: 'none',
+        color: '#949494',
+        float: 'left'
+    },
+    Arrow: {float: 'right'}
+});
+
+const calc = (x, y) => [-(y - window.innerHeight / 2) / 20, (x - window.innerWidth / 2) / 20, 1.1];
+const trans = (x, y, s) => `perspective(200px) scale(${s})`;
+
+const User = React.forwardRef(
+    ({user, isDragging, connectDragSource, connectDropTarget}, ref) => {
+        const classes = useStyles({user, isDragging, connectDragSource, connectDropTarget});
+        const elementRef = useRef(null);
+        const opacity = Number(!isDragging);
+	    const [props, set] = useSpring(() => ({ xys: [0, 0, 1], config: { mass: 5, tension: 250, friction: 20 } }));
+	    connectDragSource(elementRef);
+        connectDropTarget(elementRef);
+        useImperativeHandle(ref, () => ({
+            getNode: () => elementRef.current,
+        }));
+        return (
+	        <animated.div className={classes.Container}
+		        onMouseDown={({ clientX: x, clientY: y }) => set({ xys: calc(x, y) })}
+		        onMouseUp={() => set({ xys: [0, 0, 1] })}
+		        onMouseOver={({ clientX: x, clientY: y }) => set({ xys: calc(x, y) })}
+		        onMouseLeave={() => set({ xys: [0, 0, 1] })}
+		        onDragOver={({ clientX: x, clientY: y }) => set({ xys: calc(x, y) })}
+		        onDragLeave={() => set({ xys: [0, 0, 1] })}
+		        style={{ transform: props.xys.interpolate(trans) }}>
```

Comment:
👌 `react-spring` seems really easy to use. Have you tried some other libs?

Reply:
yeah i'd also tried `react-motion` that attempt had his own branch, that i realise i didn't pushed on github 😅

Reply:
but i have two other attempt branch